### ISSUE TITLE
Overfitting in the fitness function

### ISSUE THREAD

**Issue by JakobStadlhuber** [ultralytics/yolov5#7730]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello,

as far as I know, the fitness function is used for e.g. Early Stop and to choose the best model.

https://github.com/ultralytics/yolov5/blob/417a2f425cd553c1f2a41374a6cd2710aa91d5f0/utils/metrics.py#L13-L17

This function does not consider train loss like box, cls or obj loss, couldn't that lean to overfitting without considering these loss functions? Especially with Early Stop disabled, the best model could then be overfitted.

**Comment by glenn-jocher** [comment]
@JakobStadlhuber train losses will always trend toward zero the longer you train, they'll never show indications of overfitting. Their only purpose is for computing gradients on the model parameters which the optimizer uses to update the parameters. 

val losses or performance metrics can both be used as indicators of overfitting, what to use there is somewhat a matter of preference/domain etc.

**Comment by JakobStadlhuber** [comment]
Thanks for clarifying.

**Comment by yolor2** [comment]
@glenn-jocher how could we infer overfitting from validation loss and performance mterics graphs? could you please enlighten me. Thank you

**Comment by glenn-jocher** [comment]
@yolor2 overfitting is just increasing validation loss, i.e. so if you look at your training val losses and you see them start to increase rather than decrease then your model is overfitting your training data.
